samples/maps/demo/data-class-two-ranges/demo.js

```diff
@@ -14,24 +14,52 @@
         parsed: function (columns) {
 
             /**
-             * Event handler for clicking points. Use jQuery UI to pop up
-             * a pie chart showing the details for each state.
+             * Event handler for clicking points.
              */
             function pointClick() {
-                const row = this.options.row,
-                    $div = $('<div></div>')
-                        .dialog({
-                            title: this.name,
-                            width: 320,
-                            height: 300
-                        });
-
-                window.chart = new Highcharts.Chart({
+                const row = this.options.row;
+                const chart = this.series.chart;
+
+                // Remove existing annotation if present
+                chart.removeAnnotation('election-popup');
+
+                // Add new annotation with a pie chart
+                chart.addAnnotation({
+                    id: 'election-popup',
+                    labelOptions: {
+                        useHTML: true,
+                        backgroundColor: '#fff'
+                    },
+                    labels: [{
+                        point: {
+                            x: chart.plotWidth / 2,
+                            y: chart.plotHeight / 10
+                        },
+                        text: `
+                            <div id="annotation-header">
+                                <span>${this.name}</span>
+                                <button id="annotation-close-btn">
+                                X
+                                </button>
+                            </div>
+                            <div id="popup-pie"></div>
+                        `,
+                        shape: 'rect'
+                    }],
+                    zIndex: 10
+                });
+
+                console.log(chart);
```

Comment:
There shouldn't be any console logs in the code.

Reply:
removed